nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -69,6 +79,37 @@ public static synchronized UserGroupInformation loginKerberos(final Configuratio
         return UserGroupInformation.getCurrentUser();
     }
 
+    public static synchronized UserGroupInformation loginKerberosWithPassword(final Configuration config, final String principal, final String password) throws IOException {
```

Comment:
I think this unused at this point since the KerberoUser's are created outside and everything goes through getUgiFromKerberosUser, but I'm not against keeping the method if you want.

Reply:
Agreed on removing it.  We don't want, at this point anyway, to get a UGI without being able to associate the KerberosUser from which it was created.  I removed the method.